app/services/manifest_builder_v3.rb

```diff
@@ -0,0 +1,393 @@
+# frozen_string_literal: true
+class ManifestBuilderV3
+  attr_reader :resource, :services
+
+  ##
+  # @param [Resource] resource the Resource subject
+  def initialize(resource, auth_token = nil, current_ability = nil)
+    @resource = RootNode.for(resource, auth_token, current_ability)
+  end
+
+  ##
+  # Build the JSON-serialized Manifest instance
+  # @return [JSON]
+  def build
+    JSON.parse(manifest.to_json, symbolize_keys: true)
+  end
+
+  ##
+  # Presenter modeling the Resource subjects as root nodes
+  class RootNode
+    def self.for(resource, _auth_token = nil, _current_ability = nil)
+      case resource
+      when ScannedMap
+        ScannedMapNode.new(resource)
+      end
+    end
+    attr_reader :resource, :auth_token, :current_ability
+    delegate :decorate, :to_model, :id, to: :resource
+
+    def source_metadata_identifier
+      resource.try(:source_metadata_identifier)
+    end
+
+    ##
+    # @param [Resource] resource the Resource being modeled as the root
+    def initialize(resource, auth_token = nil, current_ability = nil)
+      @resource = resource
+      @auth_token = auth_token
+      @current_ability = current_ability
```

Comment:
Question: This seems to be the case now too, but even if I don't have permission to a child resource it'll get added to the manifest?

Reply:
Yes, it's the case now. I can't imagine, though, a case where someone would have permission on the parent and not the child.